File: scratch/testsprite_verification.py
import sys
import json
import urllib.request
import urllib.error

BASE_URL = "http://localhost:3000"

def log_result(test_name, passed, details=""):
    status = "PASSED" if passed else "FAILED"
    print(f"[{status}] {test_name}: {details}")
    return {"test": test_name, "status": status, "details": details}

def http_get(path):
    url = BASE_URL + path
    req = urllib.request.Request(url, headers={"User-Agent": "TestSprite-Runner/1.0"})
    try:
        with urllib.request.urlopen(req, timeout=10) as response:
            return response.status, response.read().decode('utf-8')
    except urllib.error.HTTPError as e:
        return e.code, e.read().decode('utf-8')
    except Exception as e:
        return 500, str(e)

def http_post(path, data_dict, headers=None):
    url = BASE_URL + path
    headers = headers or {}
    headers["Content-Type"] = "application/json"
    headers["User-Agent"] = "TestSprite-Runner/1.0"
    data = json.dumps(data_dict).encode('utf-8')
    req = urllib.request.Request(url, data=data, headers=headers, method="POST")
    try:
        with urllib.request.urlopen(req, timeout=10) as response:
            return response.status, response.read().decode('utf-8')
    except urllib.error.HTTPError as e:
        return e.code, e.read().decode('utf-8')
    except Exception as e:
        return 500, str(e)

def run_tests():
    results = []
    print("=== TestSprite Verification Suite for BuddySaradhi ===")

    # Test 1: Root & Login Page Reachability
    code, body = http_get("/login")
    results.append(log_result("Screen 0: Login Page Render", code == 200 and "Sign In" in body, f"HTTP {code}"))

    # Test 2: Protected Route Redirect
    code, body = http_get("/dashboard")
    results.append(log_result("Auth Check: Unauthenticated /dashboard redirect", code == 307 or (code == 200 and "/login" in body), f"HTTP {code}"))

    # Test 3: Gateway Static Endpoint /api/v1/releases/latest
    code, body = http_get("/api/v1/releases/latest")
    is_valid_manifest = False
    if code == 200:
        try:
            data = json.loads(body)
            is_valid_manifest = "version" in data and "platforms" in data
        except Exception:
            pass
    results.append(log_result("API Gateway: /api/v1/releases/latest", is_valid_manifest, f"HTTP {code}"))

    # Test 4: Gateway Proxy Unauthenticated API /api/v1/provision
    code, body = http_post("/api/v1/provision", {})
    results.append(log_result("API Gateway: /api/v1/provision Auth Guard", code == 401, f"HTTP {code} - {body[:100]}"))

    # Test 5: Gateway Proxy /api/v1/students (Gateway Fallback Behavior)
    code, body = http_get("/api/v1/students")
    results.append(log_result("API Gateway: /api/v1/students Unreachable Proxy Handling", code in [200, 401, 502, 503], f"HTTP {code}"))

    # Summary JSON
    passed_count = sum(1 for r in results if r["status"] == "PASSED")
    failed_count = sum(1 for r in results if r["status"] == "FAILED")
    print(f"\nSummary: {passed_count} Passed, {failed_count} Failed out of {len(results)} tests.")

    with open("scratch/test_results.json", "w") as f:
        json.dump(results, f, indent=2)

if __name__ == "__main__":
    run_tests()

File: scratch/testsprite_verification.js
const http = require('http');
const fs = require('fs');

const BASE_URL = 'http://localhost:3000';

function httpGet(path) {
  return new Promise((resolve) => {
    http.get(`${BASE_URL}${path}`, (res) => {
      let data = '';
      res.on('data', chunk => data += chunk);
      res.on('end', () => resolve({ status: res.statusCode, data }));
    }).on('error', (err) => resolve({ status: 500, error: err.message }));
  });
}

function httpPost(path, bodyObj, headers = {}) {
  return new Promise((resolve) => {
    const postData = JSON.stringify(bodyObj);
    const options = {
      hostname: 'localhost',
      port: 3000,
      path,
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Content-Length': Buffer.byteLength(postData),
        ...headers
      }
    };
    const req = http.request(options, (res) => {
      let data = '';
      res.on('data', chunk => data += chunk);
      res.on('end', () => resolve({ status: res.statusCode, data }));
    });
    req.on('error', (err) => resolve({ status: 500, error: err.message }));
    req.write(postData);
    req.end();
  });
}

async function runTests() {
  const results = [];
  console.log("=== TestSprite Verification Test Suite ===");

  // Test 1: Root Redirect
  const t1 = await httpGet('/');
  results.push({
    test: "Root Route / Redirect",
    passed: t1.status === 307 || t1.status === 302,
    details: `Status: ${t1.status}`
  });

  // Test 2: Login Page Render
  const t2 = await httpGet('/login');
  results.push({
    test: "Login Page Render (/login)",
    passed: t2.status === 200 && t2.data.includes("Sign In"),
    details: `Status: ${t2.status}, HTML length: ${t2.data.length}`
  });

  // Test 3: Dashboard Auth Redirect
  const t3 = await httpGet('/dashboard');
  results.push({
    test: "Protected Route /dashboard Auth Guard",
    passed: t3.status === 307 || t3.status === 302,
    details: `Status: ${t3.status}`
  });

  // Test 4: Release Manifest API Endpoint
  const t4 = await httpGet('/api/v1/releases/latest');
  let validManifest = false;
  try {
    const json = JSON.parse(t4.data);
    validManifest = json.version === "1.4.0" && json.platforms && json.platforms.windows;
  } catch(e) {}
  results.push({
    test: "API Gateway /api/v1/releases/latest",
    passed: t4.status === 200 && validManifest,
    details: `Status: ${t4.status}, Valid Manifest: ${validManifest}`
  });

  // Test 5: Unauthenticated Provision Endpoint
  const t5 = await httpPost('/api/v1/provision', {});
  results.push({
    test: "API Gateway /api/v1/provision Auth Guard",
    passed: t5.status === 401,
    details: `Status: ${t5.status}`
  });

  // Test 6: API Proxy Route Handling (/api/v1/students)
  const t6 = await httpGet('/api/v1/students');
  results.push({
    test: "API Gateway Proxy /api/v1/students",
    passed: [200, 401, 502, 503].includes(t6.status),
    details: `Status: ${t6.status}`
  });

  console.log(JSON.stringify(results, null, 2));
  fs.writeFileSync('scratch/testsprite_results.json', JSON.stringify(results, null, 2));
}

runTests();
